hepcrawl/spiders/cds_spider.py

```diff
@@ -138,5 +138,6 @@ def _parsed_item_from_marcxml(
                 record_format='hep',
                 exception=repr(e),
                 traceback=tb,
-                source_data=marcxml_record
+                source_data=marcxml_record,
+                file_name="__missing__"
```

Comment:
`None` doesn't work?

Reply:
file_name on ParsedItem is used in few places. It's safer to set something in there.

Reply:
What if it tries to retrieve from somewhere, isn't it safer just to use None?

Reply:
It's not retrieving it from anywhere, as it's a filename extracted from url (which is not available in this context) so it cannot be accessed locally on spider. 
I don't like it as I'm not sure what might happen in this spider when file_name will be None. In few places we are processing it as it's a string, not expecting it to be None so probably it will throw more exceptions when it will be None.

Reply:
let's just put `''` then, I see it's been used in few places. At least if there is a `if file_name
` it will be `False`

Reply:
I checked few more places in code. I will try with None. I don't like it but it will be cleaner than empty string IMHO.